test.js

```diff
@@ -244,3 +246,87 @@ test('sync (not found, custom cwd)', t => {
 
 	t.is(filePath, null);
 });
+
+test('async (custom function)', async t => {
+	const cwd = process.cwd();
+
+	t.is(await m(dir => {
+		t.is(dir, cwd);
+		return dir;
+	}), cwd);
+
+	t.is(await m(() => {
+		return '.';
+	}), cwd);
+
+	t.is(await m(() => {
+		return Promise.resolve('foo.txt');
+	}), path.join(cwd, 'foo.txt'));
```

Comment:
This can use an async function instead of `Promise.resolve`. Would be the same.

Reply:
Done. My intention was to show that it works both with an `async` function or one that returns a `Promise` manually. But I agree this doesn't have much value in the tests. Would belong more in the documentation if we did want to keep it somewhere for explanatory purposes.